lib/chef/cookbook/gem_installer.rb

```diff
@@ -48,7 +48,13 @@ def install
           begin
             Dir.mktmpdir("chef-gem-bundle") do |dir|
               File.open("#{dir}/Gemfile", "w+") do |tf|
-                tf.puts "source '#{Chef::Config[:rubygems_url]}'"
+                if Chef::Config[:rubygems_url].empty?
+                  tf.puts "source 'https://www.rubygems.org'"
+                else
+                  [ Chef::Config[:rubygems_url] ].flatten.compact.each do |s|
```

Comment:
The flatten-and-compact dance is probably not worth the CPU time here, `Array(Chef::Config[:rubygems_url] || 'https://www.rubygems.org').each` handles both conditions?